src/com/goide/psi/impl/imports/GoReferenceImporter.java

```diff
@@ -0,0 +1,87 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.psi.impl.imports;
+
+import com.goide.GoLanguage;
+import com.goide.codeInsight.imports.GoCodeInsightSettings;
+import com.goide.codeInsight.imports.GoImportPackageQuickFix;
+import com.goide.psi.GoElement;
+import com.goide.psi.impl.GoReference;
+import com.intellij.codeInsight.daemon.ReferenceImporter;
+import com.intellij.codeInsight.daemon.impl.CollectHighlightsUtil;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiReference;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class GoReferenceImporter implements ReferenceImporter {
+  @Override
+  public boolean autoImportReferenceAtCursor(@NotNull Editor editor, @NotNull PsiFile file) {
+    if (!file.getViewProvider().getLanguages().contains(GoLanguage.INSTANCE) || !GoCodeInsightSettings.getInstance().ADD_UNAMBIGUOUS_IMPORTS_ON_THE_FLY) {
+      return false;
+    }
+
+    int caretOffset = editor.getCaretModel().getOffset();
+    Document document = editor.getDocument();
+    int lineNumber = document.getLineNumber(caretOffset);
+    int startOffset = document.getLineStartOffset(lineNumber);
+    int endOffset = document.getLineEndOffset(lineNumber);
+
+    List<PsiElement> elements = CollectHighlightsUtil.getElementsInRange(file, startOffset, endOffset);
+    for (PsiElement element : elements) {
+      if (element instanceof GoElement) {
+        for (PsiReference reference : element.getReferences()) {
+          if (reference instanceof GoReference) {
+            GoImportPackageQuickFix fix = new GoImportPackageQuickFix(reference);
+            if (fix.isAvailable(file.getProject(), editor, file)) {
+              new GoImportPackageQuickFix(reference).applyFix();
+              return true;
+            }
+          }
+        }
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean autoImportReferenceAt(@NotNull Editor editor, @NotNull PsiFile file, int offset) {
+    if (!file.getViewProvider().getLanguages().contains(GoLanguage.INSTANCE) || !GoCodeInsightSettings.getInstance().ADD_UNAMBIGUOUS_IMPORTS_ON_THE_FLY) {
+      return false;
+    }
+
+    final PsiReference reference = file.findReferenceAt(offset);
+    if (reference instanceof GoReference) {
+      GoImportPackageQuickFix fix = new GoImportPackageQuickFix(reference);
+      if (fix.isAvailable(file.getProject(), editor, file)) {
+        WriteCommandAction.runWriteCommandAction(file.getProject(), new Runnable() {
+          @Override
+          public void run() {
+            new GoImportPackageQuickFix(reference).applyFix();
```

Comment:
why do you need second quickFix?
are you sure that fix is still applicable when write actions is started?